src/realm/sync/subscriptions.cpp

```diff
@@ -314,6 +373,10 @@ void SubscriptionSet::commit()
         update_state(State::Pending, util::none);
     }
     m_tr->commit_and_continue_as_read();
+
+    if (m_state_changed) {
```

Comment:
I guess right now it will always be true. When I was writing this, I wanted to distinguish between a mutable subscription set being mutated by the sync client (i.e. it's state is changing because a download message or something was received) vs being mutated by an SDK, but right now the former will always change the state, and the latter only happens when the state is transitioning from Uncommitted to Pending.

Reply:
I'd recommend only adding it later if needed.

Reply:
Done.